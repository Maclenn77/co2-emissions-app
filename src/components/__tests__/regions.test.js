/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { HashRouter as Router } from 'react-router-dom';
import Region from '../region';

const mockStore = configureMockStore();

const store = mockStore({
  co2Emissions: {
    0: {
      id: 0,
      iso: 'mex',
      label: 'Mexico',
      date: 2018,
      co2Emissions: 6.0000,
    },
  },
});

it('renders correctly', () => {
  const mockRegion = {
    iso: 'MCK',
    label: 'Mock Region',
    date: 2018,
    co2Emissions: 6.0000,
  };
  const component = renderer.create(
    <Provider store={store}>
      <Router>
        <Region region={mockRegion} />
      </Router>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
