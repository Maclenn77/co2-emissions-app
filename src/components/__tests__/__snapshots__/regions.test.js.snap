// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  href="#/countries/MCK"
  onClick={[Function]}
>
  <div
    className="p-2"
  >
    <div
      className="arrow-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M14 8.83L17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
        />
      </svg>
    </div>
    <h2>
      Mock Region
    </h2>
    <h6
      className="co2"
    >
      CO
      <sub>
        2
      </sub>
      :
    </h6>
    <p>
      6
    </p>
  </div>
</a>
`;
